network/network_windows.go

```diff
@@ -347,6 +395,13 @@ func (nm *networkManager) newNetworkImplHnsV2(nwInfo *NetworkInfo, extIf *extern
 		logger.Info("Network with name already exists", zap.String("name", hcnNetwork.Name))
 	}
 
+	// check if ipv6 default gateway route is missing before windows endpoint creation
+	if len(nwInfo.Subnets) >= numDualStackSubnet {
+		if err = nm.addIPv6DefaultRoute(); err != nil {
+			return nil, errors.Wrapf(err, "failed to add missing ipv6 default route to windows node active and persistent store")
+		}
+	}
+
```

Comment:
did we test if hns network succeeded but adding route failed. on second call everything should succeed, since we are not cleaning up hns network if adding ipv6 route fails. lets also add UT for that specific case